context.go

```diff
@@ -336,38 +336,62 @@ func (c *Context) Rem(d, x, y *Decimal) (Condition, error) {
 }
 
 // Sqrt sets d to the square root of x.
+// Sqrt uses the Babylonian method for computing the square root, which uses
+// O(log p) steps for p digits of precision. We follow the algorithm in
+// Properly Rounded Variable Precision Square Root by T. E. Hull and A. Abrham,
+// ACM Transactions on Mathematical Software, Vol 11 #3, pp229–237, ACM, September 1985.
 func (c *Context) Sqrt(d, x *Decimal) (Condition, error) {
-	// See: Properly Rounded Variable Precision Square Root by T. E. Hull
-	// and A. Abrham, ACM Transactions on Mathematical Software, Vol 11 #3,
-	// pp229–237, ACM, September 1985.
 
 	switch x.Sign() {
 	case -1:
-		res := InvalidOperation
-		return c.goError(res)
+		return c.goError(InvalidOperation)
 	case 0:
 		d.SetCoefficient(0)
 		d.Exponent = 0
 		return 0, nil
 	}
-
-	// Use same precision as in decNumber.
-	workp := c.Precision + 1
-	if nd := uint32(x.NumDigits()); workp < nd {
-		workp = nd
+	// x > 0.
+
+	// TODO(mjibson): Also it would be good if we could keep the number of
+	// variables in scope low. This refactoring doesn't quite work, but perhaps
+	// you can refactor the calls below to use nc.Precision instead of keeping
+	// workp in scope throughout.
+	// TODO(mjibson): Also, what's nc?
+	var nc *Context
+	{
+		// workp is the number of digits of precision used.
+		// We use same precision as in decNumber.
+		workp := c.Precision + 1
+		if nd := uint32(x.NumDigits()); workp < nd {
+			workp = nd
+		}
+		if workp < 7 {
+			workp = 7
+		}
+		nc = c.WithPrecision(workp)
 	}
-	if workp < 7 {
-		workp = 7
+
+	// TODO(mjibson): Too many variables without comments/definitions. And nd is a temporary variable that goes out of scope, perhaps it can be closed off? I've given it a try but I'm not too happy with this either... Also, these are my *guesses* for what these variables represent.
```

Comment:
`nd` is the only variable I can see as benefiting from this, but it doesn't even gain that much because it requires a `var e blah` above it, which I don't think is better. All the others are used throughout this function because that's how they work in the paper or decNumber.